### PR TITLE
Store backup and restore

### PR DESCRIPTION
**Pull request by jakexks**
This adds the ability to create snapshots of go-micro stores, and restore snapshots to go-micro stores.

### CODE REVIEW

File: store/snapshot/snapshotter.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	"github.com/micro/go-micro/v2/store"
+	"github.com/pkg/errors"
+)
+
+// Snapshotter creates snapshots of a go-micro store
+type Snapshotter interface {
+	// Init validates the Snapshotter options and returns an error if they are invalid.
+	// Init must be called before the Snapshotter is used
+	Init(opts ...SnapshotterOption) error
```
Comment (crufter) [inline store/snapshot/snapshotter.go:18]: Why not have this in the constructor of a snapshotter? Custom implementations might use additional options etc anyway.

Comment (jakexks) [reply store/snapshot/snapshotter.go:18]: Yeah, wasn't sure if this was the best interface

File: store/snapshot/snapshotter.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	"os"
+	"sync"
+	"time"
+
+	"github.com/micro/go-micro/v2/store"
+	"github.com/pkg/errors"
+)
+
+// Snapshotter creates snapshots of a go-micro store
+type Snapshotter interface {
```
Comment (asim) [inline store/snapshot/snapshotter.go:15]: Call it Snapshot or Backup

File: store/snapshot/snapshotter.go (hunk 1 above)
Comment (asim) [reply store/snapshot/snapshotter.go:18]: It's fine for now. It's internal, works with our store and the usage is CLI right. So what we want is `micro store snapshot/restore`

**Pull request merged**